mlflow/tensorflow/callback.py

```diff
@@ -0,0 +1,87 @@
+from tensorflow import keras
+
+from mlflow import log_param, log_text
+from mlflow.utils.autologging_utils import BatchMetricsLogger
+
+
+class MLflowLoggingCallback(keras.callbacks.Callback):
+    """Callback for logging Tensorflow training metrics to MLflow.
+
+    This callback logs training metrics every epoch or every n steps (defined by the user) to
+    MLflow.
+
+    Args:
+        run: an 'mlflow.entities.run.Run' instance, the MLflow run.
+        log_every_epoch: bool, If True, log metrics every epoch. If False, log metrics every n
+            steps.
+        log_every_n_steps: int, log metrics every n steps. If None, log metrics every epoch.
+
+    Examples:
+        ```python
+        from tensorflow import keras
+        import mlflow
+        import numpy as np
+
+        # Prepare data for a 2-class classification.
+        data = tf.random.uniform([8, 28, 28, 3])
+        label = tf.convert_to_tensor(np.random.randint(2, size=8))
+
+        model = keras.Sequential(
+            [
+                keras.Input([28, 28, 3]),
+                keras.layers.Flatten(),
+                keras.layers.Dense(2),
+            ]
+        )
+
+        model.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(0.001),
+            metrics=[keras.metrics.SparseCategoricalAccuracy()],
+        )
+
+        with mlflow.start_run() as run:
+            model.fit(
+                data,
+                label,
+                batch_size=4,
+                epochs=2,
+                callbacks=[MLflowMetricsLoggingCallback(run)],
+            )
+        ```
+    """
+
+    def __init__(self, run, log_every_epoch=True, log_every_n_steps=None):
```

Comment:
```suggestion
    def __init__(self, run_id, log_every_epoch=True, log_every_n_steps=None):
```

Can we use `run_id` instead of `run`?

Reply:
honestly I have been back and forth about `run_id` vs `run` in the constructor. I think `run` is slightly better because users can just do `callback = MLflowCallback(run)` instead of `callback = MLflowCallback(run.info.run_id)`, we are hiding the complexity under the hood.

Reply:
Makes sense, let's use `run`.